torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -419,19 +426,105 @@ def generate_plot_visualization(self, feature_filter: str, module_fqn_filter: st
 
         Note:
             Only features in the report that have tensor value data are plottable by this class
-
+            When the tensor information is plotted, it will plot:
+                idx as the x val, feature value as the y_val
+            When the channel information is plotted, it will plot:
+                idx // 3 as the x val, feature value as the y_val [for each channel]
+                The reason for this is that we want to be able to compare values across the
+                channels for same layer, and it will be hard if values are staggered by idx
         Args:
-            feature_filter (str, optional): Filters the features presented to only those that
+            feature_filter (str): Filters the features presented to only those that
                 contain this filter substring
-                Default = "", results in all the features being printed
             module_fqn_filter (str, optional): Only includes modules that contains this string
                 Default = "", results in all the modules in the reports to be visible in the table
 
         Expected Use:
             >>> # the code below both returns the info and diplays the plot
             >>> model_report_visualizer.generate_plot_visualization(*filters) # plots the data
         """
-        pass
+        # checks if we have matplotlib and let's user know to install it if don't
+        if not got_matplotlib:
+            print("make sure to install matplotlib and try again.")
+            return None
+
+        # get the table dict and the specific tables of interest
+        table_dict = self.generate_filtered_tables(feature_filter, module_fqn_filter)
+        tensor_headers, tensor_table = table_dict[self.TABLE_TENSOR_KEY]
+        channel_headers, channel_table = table_dict[self.TABLE_CHANNEL_KEY]
+
+        # make sure it is only 1 feature that is being plotted
+        # get the number of features in each of these
+        tensor_info_features_count = len(tensor_headers) - ModelReportVisualizer.DEFAULT_NON_FEATURE_TENSOR_HEADERS
+        channel_info_features_count = len(channel_headers) - ModelReportVisualizer.DEFAULT_NON_FEATURE_CHANNEL_HEADERS
+
+        # keep track of per_channel or not
+        data_is_per_channel: bool = False
+
+        x_data: List = []
+        y_data: List[List] = []
+        # the feature will either be a tensor feature or channel feature
+        if tensor_info_features_count == 1:
```

Comment:
can we just plot multiple figures if there are multiple selected features?

Reply:
since the plotting function is pretty straightforward, we just decided to just call separately for each thing they want to plot rather than build that functionality into the code itself